src/dhtproto/client/legacy/internal/helper/NodeAvailability.d

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+
+    TODO
+
+    Copyright:
+        Copyright (c) 2018 dunnhumby Germany GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module dhtproto.client.legacy.internal.helper.NodeAvailability;
+
+public struct NodeAvailabilty
+{
+    import swarm.Const;
+    import core.sys.posix.time;
+    import dhtproto.client.legacy.internal.registry.model.IDhtNodeRegistryInfo;
+
+    /// TODO
+    private struct NodeErrorTracker
+    {
+        /// TODO
+        private time_t last_error_timestamp;
+
+        /// TODO
+        public void error ( )
+        {
+            this.last_error_timestamp = time();
+        }
+
+        /// TODO
+        public bool available ( )
+        {
+            const delay_ms = 3_000; // TODO: configurable
+            auto diff = time() - this.last_error_timestamp;
+            return diff >= delay_ms;
```

Comment:
You're comparing seconds with milliseconds here.

I suggest you change `time` to return the time in milliseconds:

`return ts.tv_sec + ts.tv_nsec / 1_000_000`.

Reply:
Haha yes. I've not tested this yet ;)